# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/nr_socket_tcp.cpp
# Commit: b879a0044cf3
# Full Hash: b879a0044cf39f0230879d075c915485affb7707
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Teach WebrtcTCPSocket to bind to a specific host/port. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45102
# ==============================================================================

diff -r f5353ea9f1df -r b879a0044cf3 media/mtransport/nr_socket_tcp.cpp
--- a/media/mtransport/nr_socket_tcp.cpp	Tue Sep 17 18:17:14 2019 +0000
+++ b/media/mtransport/nr_socket_tcp.cpp	Tue Sep 17 18:17:18 2019 +0000
@@ -97,16 +97,26 @@
 int NrTcpSocket::connect(nr_transport_addr* aAddr) {
   r_log(LOG_GENERIC, LOG_DEBUG, "NrTcpSocket::Connect %p\n", this);
 
-  nsCString host;
-  int port;
+  nsCString remote_addr;
+  int remote_port;
 
-  if (nr_transport_addr_get_addrstring_and_port(aAddr, &host, &port)) {
+  if (NS_WARN_IF(nr_transport_addr_get_addrstring_and_port(aAddr, &remote_addr,
+                                                           &remote_port))) {
+    return R_FAILED;
+  }
+
+  nsCString local_addr;
+  int local_port;
+
+  if (NS_WARN_IF(nr_transport_addr_get_addrstring_and_port(
+          &my_addr_, &local_addr, &local_port))) {
     return R_FAILED;
   }
 
   mWebrtcTCPSocket = new WebrtcTCPSocketWrapper(this);
 
-  mWebrtcTCPSocket->AsyncOpen(host, port, mConfig);
+  mWebrtcTCPSocket->AsyncOpen(remote_addr, remote_port, local_addr, local_port,
+                              mConfig);
 
   // trigger nr_socket_buffered to set write/read callback
   return R_WOULDBLOCK;
